server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -379,64 +378,79 @@ static boolean executeBulkItemRequest(
                 request.getAutoGeneratedTimestamp(),
                 request.isRetry()
             );
-
+            if (result.getResultType() == Engine.Result.Type.MAPPING_UPDATE_REQUIRED) {
+                return handleMappingUpdateRequired(
+                    context,
+                    mappingUpdater,
+                    waitForMappingUpdate,
+                    itemDoneListener,
+                    primary,
+                    result,
+                    version,
+                    updateResult
+                );
+            }
         }
-        if (result.getResultType() == Engine.Result.Type.MAPPING_UPDATE_REQUIRED) {
-
-            try {
-                Optional<CompressedXContent> mergedSource = Optional.ofNullable(
-                    primary.mapperService()
-                        .merge(
-                            MapperService.SINGLE_MAPPING_NAME,
-                            new CompressedXContent(result.getRequiredMappingUpdate()),
-                            MapperService.MergeReason.MAPPING_AUTO_UPDATE_PREFLIGHT
-                        )
-                ).map(DocumentMapper::mappingSource);
-                Optional<CompressedXContent> previousSource = Optional.ofNullable(primary.mapperService().documentMapper())
-                    .map(DocumentMapper::mappingSource);
+        onComplete(result, context, updateResult);
```

Comment:
The `onComplete` change: we call it the same way as before, just no longer from an else block. Effectively, this improves readability and is not a functional change, correct?

Reply:
Exactly it's the same code just a little drier :)